internal/controller/nodeimageset_controller.go

```diff
@@ -68,6 +69,10 @@ func (r *NodeImageSetReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			if err := r.Update(ctx, &nodeImageSet); err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete all metrics for this NodeImageSet
+			metrics.ImageInfoVec.DeletePartialMatch(map[string]string{"name": nodeImageSet.Name})
+			metrics.ImageSizeBytesVec.DeletePartialMatch(map[string]string{"name": nodeImageSet.Name})
+			metrics.ImagePrefetchDurationSecondsVec.DeletePartialMatch(map[string]string{"name": nodeImageSet.Name})
```

Comment:
These should be done before removing the finalizer.
If an error occurs during the update, metrics may leak depending on the error.

Reply:
Thank you.
I fixed the code to ensure that deleting metrics is done before removing the finalizer.